tests/basic/misc_test.go

```diff
@@ -1157,3 +1161,102 @@ var _ = Describe("{AutoFSTrimReplAddWithNoPool0}", func() {
 		AfterEachTest(contexts, testrailID, runID)
 	})
 })
+
+// Add a test to detach all the disks from a node and then reattach them to a different node
+var _ = Describe("{NodeDiskDetachAttach}", func() {
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("NodeDiskDetachAttach", "Validate disk detach and attach", nil, 0)
+	})
+	var contexts []*scheduler.Context
+
+	testName := "nodediskdetachattach"
+	stepLog := "has to detach all disks from a node and reattach them to a different node"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("%s-%d", testName, i))...)
+		}
+
+		randomNum := rand.New(rand.NewSource(time.Now().Unix()))
+
+		ValidateApplications(contexts)
+		// Fetch the node where PX is not started
+		nonPXNodes := node.GetPXDisabledNodes()
+		// Remove disks from all nodes in nonPXNodes
+		for _, n := range nonPXNodes {
+			// Fetch the disks attached to the node
+			err := Inst().N.RemoveNonRootDisks(n)
+			log.FailOnError(err, fmt.Sprintf("Failed to remove disks on node %s", n.Name))
+		}
+		randNonPxNode := nonPXNodes[randomNum.Intn(len(nonPXNodes))]
+
+		// Fetch a random node from the StorageNodes
+		storageNodes := node.GetStorageNodes()
+		randomStorageNode := storageNodes[randomNum.Intn(len(storageNodes))]
+
+		// Fetch random storage node and detach the disks attached to that node
+		var oldNodeIDtoMatch string
+		var newNodeIDtoMatch string
+		Step(fmt.Sprintf("detach disks attached from a random storage node %s and attach to non px node %s", randomStorageNode.Name, randNonPxNode.Name), func() {
+			// ToDo - Ensure the above selected node has volumes/apps running on it
+			// Store the Node ID of randomStorageNode for future use
+			oldNodeIDtoMatch = randomStorageNode.Id
+			// Stop PX on the node
+			err = k8sCore.AddLabelOnNode(randomStorageNode.Name, schedops.PXServiceLabelKey, "stop")
```

Comment:
you can use existing function StopPxOnNode

Reply:
will take care of it as part of PTX-23734